src/Symfony/Component/HtmlSanitizer/HtmlSanitizerConfig.php

```diff
@@ -0,0 +1,373 @@
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HtmlSanitizer;
+
+use Symfony\Component\HtmlSanitizer\Reference\W3CReference;
+use Symfony\Component\HtmlSanitizer\Visitor\AttributeSanitizer\AttributeSanitizerInterface;
+
+/**
+ * @author Titouan Galopin <galopintitouan@gmail.com>
+ *
+ * @experimental
+ */
+class HtmlSanitizerConfig
+{
+    /**
+     * Elements that should be removed but their children should be retained.
+     *
+     * @var array<string, bool>
+     */
+    private array $blockedElements = [];
+
+    /**
+     * Elements that should be retained, with their allowed attributes.
+     *
+     * @var array<string, array<string, bool>>
+     */
+    private array $allowedElements = [];
+
+    /**
+     * Links schemes that should be retained, other being dropped.
+     *
+     * @var string[]
+     */
+    private array $allowLinkSchemes = ['http', 'https', 'mailto', 'tel'];
+
+    /**
+     * Links hosts that should be retained (by default, all hosts are allowed).
+     *
+     * @var string[]|null
+     */
+    private ?array $allowLinkHosts = null;
+
+    /**
+     * Should the sanitizer allow relative links (by default, they are dropped).
+     */
+    private bool $allowRelativeLinks = false;
+
+    /**
+     * Image/Audio/Video schemes that should be retained, other being dropped.
+     *
+     * @var string[]
+     */
+    private array $allowMediaSchemes = ['http', 'https', 'data'];
+
+    /**
+     * Image/Audio/Video hosts that should be retained (by default, all hosts are allowed).
+     *
+     * @var string[]|null
+     */
+    private ?array $allowMediaHosts = null;
+
+    /**
+     * Should the sanitizer allow relative media URL (by default, they are dropped).
+     */
+    private bool $allowRelativeMedias = false;
+
+    /**
+     * Should the URL in the sanitized document be transformed to HTTPS if they are using HTTP.
+     */
+    private bool $forceHttpsUrls = false;
+
+    /**
+     * Sanitizers that should be applied to specific attributes in addition to standard sanitization.
+     *
+     * @var Visitor\AttributeSanitizer\AttributeSanitizerInterface[]
+     */
+    private array $attributesSanitizers;
+
+    public function __construct()
+    {
+        $this->attributesSanitizers = [
+            new Visitor\AttributeSanitizer\AHrefSanitizer(),
+            new Visitor\AttributeSanitizer\MediaSrcSanitizer(),
+        ];
+    }
+
+    /**
+     * Create a configuration prefilled with allowed elements and attributes from the
+     * W3C Sanitizer API standard. {@see https://wicg.github.io/sanitizer-api/#default-configuration}.
+     */
+    public static function createStandardConfig(): static
+    {
+        $self = new self();
+
+        foreach (array_merge(W3CReference::BODY_ELEMENTS, W3CReference::HEAD_ELEMENTS) as $element) {
```

Comment:
this us broken (see psalm detecting it too). Those constants are storing the element name as array key, not as array value.